### PR TITLE
feat: fix log crate output

### PR DESCRIPTION
**Pull request by tustvold**
A number of crates we make use of, including rust-rdkafka use the log crate instead of tokio-tracing. Currently these logs are just silently dropped, this is not ideal.

This PR installs a shim to forward log crate output to tokio-tracing, where it feeds into the filter pipeline as normal.

It also removes tracing_subscriber from observability_deps as it is not a small dependency, and does not need to be included in every crate. This should help with build times.

### CODE REVIEW

File: trogging/src/lib.rs [hunk 1]
```diff
@@ -457,6 +462,7 @@ where
     pub fn install_global(self) -> Result<TracingGuard> {
         let subscriber = self.build()?;
         tracing::subscriber::set_global_default(subscriber)?;
+        tracing_log::LogTracer::init()?;
```
Comment (tustvold) [inline trogging/src/lib.rs:465]: This is the actual change

File: src/influxdb_ioxd.rs [hunk 2]
```diff
@@ -229,7 +229,8 @@ async fn serve(
     .fuse();
     info!("HTTP server listening");
 
-    info!("InfluxDB IOx server ready");
+    // Purposefully use log not tokio-tracing to ensure correctly hooked up
+    log::info!("InfluxDB IOx server ready");
```
Comment (tustvold) [inline src/influxdb_ioxd.rs:233]: This is then used by an end-to-end test to ensure that this pipeline is correctly hooked up

**Review by mkmik — APPROVED** [review]
how is the log level for that log crate governed?

**Comment by tustvold** [comment]
> how is the log level for that log crate governed?

By tokio-tracing, it just feeds into the same filter stack

File: trogging/src/lib.rs (hunk 1 above)
Comment (alamb) [reply trogging/src/lib.rs:465]: TIL https://docs.rs/tracing-log/0.1.2/tracing_log/ 👍

**Review by alamb — APPROVED** [review]
As DataFusion also uses the `log` crate this PR will likely make DataFusion logs available to IOx as well. 👍  Thanks @tustvold

**Pull request merged**